components/brave_wallet_ui/components/desktop/hardware-wallet-connect/hardware_wallet_connect.tsx

```diff
@@ -117,205 +168,195 @@ export const HardwareWalletConnect = ({
+  const [selectedHardwareVendor, setSelectedHardwareVendor] =
+    React.useState<BraveWallet.HardwareVendor>()
+  const [isLoadingAccounts, setIsLoadingAccounts] =
+    React.useState<boolean>(false)
+  const [accounts, setAccounts] = React.useState<AccountFromDeviceListItem[]>(
+    []
+  )
   const [connectionError, setConnectionError] = React.useState<
     ErrorMessage | undefined
   >(undefined)
-  const [selectedDerivationScheme, setSelectedDerivationScheme] =
-    React.useState<HardwareDerivationScheme>(
-      selectedAccountType.coin === BraveWallet.CoinType.SOL
-        ? SolDerivationPaths.Default
-        : LedgerDerivationPaths.LedgerLive
-    )
+  const [currentDerivationScheme, setCurrentDerivationScheme] =
+    React.useState<DerivationScheme>(DerivationSchemes.EthLedgerLive)
   const [showAccountsList, setShowAccountsList] = React.useState<boolean>(false)
-  const [filecoinNetwork, setFilecoinNetwork] =
-    React.useState<FilecoinNetwork>('f')
   const [showAuthorizeDevice, setShowAuthorizeDevice] =
     React.useState<boolean>(false)
   const hideAuthorizeDevice = () => setShowAuthorizeDevice(false)
+  const [totalNumberOfAccounts, setTotalNumberOfAccounts] = React.useState(0)
 
-  // methods
-  const onFilecoinNetworkChanged = React.useCallback(
-    (network: FilecoinNetwork) => {
-      // clear previous accounts & show loading spinner
-      setAccounts([])
-      setFilecoinNetwork(network)
-      onConnectHardwareWallet({
-        hardware: BraveWallet.LEDGER_HARDWARE_VENDOR,
-        startIndex: 0,
-        stopIndex: DerivationBatchSize,
-        network: network,
-        coin: BraveWallet.CoinType.FIL,
-        onAuthorized: hideAuthorizeDevice
-      })
-        .then((result) => {
-          setAccounts(result)
-        })
-        .catch((error) => {
-          setConnectionError(getErrorMessage(error, selectedAccountType.name))
-          setShowAccountsList(false)
-        })
-        .finally(() => setIsConnecting(false))
-    },
-    [selectedAccountType]
-  )
+  const currentHardwareImportScheme: HardwareImportScheme =
+    findHardwareImportScheme(currentDerivationScheme)
 
-  const onAddHardwareAccounts = React.useCallback(
-    async (accounts: BraveWallet.HardwareWalletAccount[]) => {
-      try {
-        await importHardwareAccounts(accounts).unwrap()
-        onSuccess()
-      } catch (error) {
-        console.log(error)
+  const setHardwareImportScheme = (scheme: DerivationScheme) => {
+    if (currentDerivationScheme === scheme) {
+      return
+    }
+    setAccounts([])
+    setTotalNumberOfAccounts(DerivationBatchSize)
+    setCurrentDerivationScheme(scheme)
+  }
+
+  const onAccountChecked = (path: string, checked: boolean) => {
+    const newAccounts = accounts.map((acc) => {
+      if (acc.derivationPath === path) {
+        return { ...acc, shouldAddToWallet: checked }
       }
-    },
-    [onSuccess, importHardwareAccounts]
-  )
+      return acc
+    })
+    setAccounts(newAccounts)
```

Comment:
Can be rewritten to not depend on the `accounts` variable
```suggestion
    setAccounts(prevAccounts => prevAccounts.map((acc) => {
      if (acc.derivationPath === path) {
        return { ...acc, shouldAddToWallet: checked }
      }
      return acc
    }))
```